ocp-metadata/ocp-metadata.go

```diff
@@ -219,24 +219,26 @@ func (meta *Metadata) getNodesInfo() (nodeInfo, error) {
 		return nodeInfoData, err
 	}
 	nodeInfoData.totalNodes = len(nodes.Items)
+newNode:
 	for _, node := range nodes.Items {
```

Comment:
```suggestion
        for _, node := range nodes.Items {
            labels := node.Labels
            workerNode := true
            // Check for master role
            if _, ok := labels["node-role.kubernetes.io/master"]; ok {
	            nodeInfoData.masterCount++
	            nodeInfoData.masterType = labels["node.kubernetes.io/instance-type"]
	            continue
            }
            
            // Check for infra role
            if _, ok := labels["node-role.kubernetes.io/infra"]; ok {
	            nodeInfoData.infraCount++
	            nodeInfoData.infraType = labels["node.kubernetes.io/instance-type"]
	            continue
            }
            
            // Check for infra role
            if _, ok := labels["node-role.kubernetes.io/workload"]; ok {
	            continue
            }
            
            // Check for worker role
            if _, ok := labels["node-role.kubernetes.io/worker"]; ok && workerNode {
	            nodeInfoData.workerCount++
	            nodeInfoData.workerType = labels["node.kubernetes.io/instance-type"]
            }
        }
```

Reply:
Since the node.Labels is a map, instead of having a complex logic to loop over the keys for labels, suggesting to just have keys checks and fetch the `nodeInfoData`.

Reply:
I was pinging with Raul about this. We need a node that has both 'worker' and 'infra' labels to _not_ be counted in the worker count. The logic has gives the correct counts, and I believe the example you gave still has the bug.

Reply:
In the suggestion I have given, if a node has both `infra` and `worker` nodes, it will be counted as `infra` as expected and we will move to scan next node. Feel free to correct me or provide with example scenarios, If my understanding is incorrect.

Reply:
Good suggestion!, I pushed it with some other changes to simplify the code even more.